testsuite/forge/src/tx_emitter.rs

```diff
@@ -0,0 +1,717 @@
+        println!(
+            "Will create {} accounts_per_client with total {} accounts",
+            req.accounts_per_client, num_accounts
+        );
+        self.mint_accounts(&req, num_accounts).await?;
+        let all_accounts = self.accounts.split_off(self.accounts.len() - num_accounts);
+        let mut workers = vec![];
+        let all_addresses: Vec<_> = all_accounts.iter().map(|d| d.address()).collect();
+        let all_addresses = Arc::new(all_addresses);
+        let mut all_accounts = all_accounts.into_iter();
+        let stop = Arc::new(AtomicBool::new(false));
+        let stats = Arc::new(StatsAccumulator::default());
+        let tokio_handle = Handle::current();
+        for client in req.json_rpc_clients {
+            for _ in 0..workers_per_ac {
+                let accounts = (&mut all_accounts).take(req.accounts_per_client).collect();
+                let all_addresses = all_addresses.clone();
+                let stop = stop.clone();
+                let params = req.thread_params.clone();
+                let stats = Arc::clone(&stats);
+                let worker = SubmissionWorker {
+                    accounts,
+                    client: client.clone(),
+                    all_addresses,
+                    stop,
+                    params,
+                    stats,
+                    tx_factory: self.tx_factory.clone(),
+                };
+                let join_handle = tokio_handle.spawn(worker.run().boxed());
+                workers.push(Worker { join_handle });
+            }
+        }
+        info!("Tx emitter workers started");
+        Ok(EmitJob {
+            workers,
+            stop,
+            stats,
+        })
+    }
+
+    pub async fn stop_job(&mut self, job: EmitJob) -> TxStats {
+        job.stop.store(true, Ordering::Relaxed);
+        for worker in job.workers {
+            let mut accounts = worker
+                .join_handle
+                .await
+                .expect("TxEmitter worker thread failed");
+            self.accounts.append(&mut accounts);
+        }
+        job.stats.accumulate()
+    }
+
+    pub async fn emit_txn_for(
+        &mut self,
+        duration: Duration,
+        emit_job_request: EmitJobRequest,
+    ) -> Result<TxStats> {
+        let job = self.start_job(emit_job_request).await?;
+        println!("starting emitting txns for {} secs", duration.as_secs());
+        tokio::time::sleep(duration).await;
+        let stats = self.stop_job(job).await;
+        Ok(stats)
+    }
+
+    fn pick_mint_client<'a, 'b>(&'a self, clients: &'b [JsonRpcClient]) -> &'b JsonRpcClient {
+        let mut rng = ThreadRng::default();
+        clients
+            .choose(&mut rng)
+            .expect("json-rpc clients can not be empty")
+    }
+}
+
+const TXN_EXPIRATION_SECONDS: i64 = 150;
+const TXN_MAX_WAIT: Duration = Duration::from_secs(TXN_EXPIRATION_SECONDS as u64 + 30);
+const MAX_TXNS: u64 = 1_000_000;
+const SEND_AMOUNT: u64 = 1;
```

Comment:
Nit:  Might be better if the constants are all together at the top, so one doesn't have to search for them in the middle